source/common/tls/tls_certificate_selector_impl.cc

```diff
@@ -0,0 +1,273 @@
+#include "source/common/tls/tls_certificate_selector_impl.h"
+
+#include "source/common/tls/utility.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace TransportSockets {
+namespace Tls {
+
+DefaultTlsCertificateSelectorImpl::DefaultTlsCertificateSelectorImpl(
```

Comment:
I don't think this class needs `Impl` in the name because it's not the one and only implementation of an interface.